pkg/lib/infra/sms/client.go

```diff
@@ -38,6 +38,7 @@ type Client struct {
 	TwilioClient                 *TwilioClient
 	NexmoClient                  *NexmoClient
 	CustomClient                 *CustomClient
+	ProviderResolver             *ProviderResolver
```

Comment:
1. Remove TwilioClient, NexmoClient, CustomClient in this struct
2. Remove NewTwilioClient, NewNexmoClient, NewCustomClient from the deps.
3. Inject ClientResolver here.
4. Use ClientResolver to resolve a client to use.
5. The API of ClientResolver is `func ResolveClient() (RawClient, error)`